src/views/InvolvementProfile/components/Membership/components/AdminCard/index.js

```diff
@@ -52,20 +52,19 @@ const AdminCard = ({ createSnackbar, isSiteAdmin, involvementDescription, onAddM
   };
 
   const handleAddMember = async () => {
-    let memberEmail = username;
-    if (!memberEmail.toLowerCase().includes('@gordon.edu')) {
-      memberEmail = memberEmail + '@gordon.edu';
+    if (!username.toLowerCase().includes('@gordon.edu')) {
+      setUsername(username.replace('@gordon.edu', ''));
     }
 
     try {
       let data = {
-        ACT_CDE: involvementCode,
-        SESS_CDE: sessionCode,
-        // TODO: Fix API to accept username instead of ID and then remove Group Admin privilege to access ID.
-        ID_NUM: (await membershipService.getEmailAccount(memberEmail)).GordonID,
-        PART_CDE: participationCode,
-        COMMENT_TXT: titleComment,
-        GRP_ADMIN: false,
+        Activity: involvementCode,
+        Session: sessionCode,
+        Username: username,
```

Comment:
So you're calling `setUsername`, but then referencing `username` in the same function. There's no guarantee that `username` will have updated by the time you reference it here, which could cause serious and unpredictable issues.